pulsar-function-go/pf/context.go

```diff
@@ -25,18 +25,21 @@ import (
 	"time"
 
 	"github.com/apache/pulsar-client-go/pulsar"
+	log "github.com/apache/pulsar/pulsar-function-go/logutil"
 )
 
 // FunctionContext provides contextual information to the executing function.
 // Features like which message id we are handling, whats the topic name of the
 // message, what are our operating constraints, etc can be accessed by the
 // executing function
 type FunctionContext struct {
-	instanceConf  *instanceConf
-	userConfigs   map[string]interface{}
-	logAppender   *LogAppender
-	outputMessage func(topic string) pulsar.Producer
-	record        pulsar.Message
+	instanceConf   *instanceConf
+	userConfigs    map[string]interface{}
+	logAppender    *LogAppender
+	outputMessage  func(topic string) pulsar.Producer
+	Publish        func(topic string, payload []byte)
+	PublishMessage func(topic string, message pulsar.ProducerMessage)
```

Comment:
Can you give little more control to the caller with a context and return an error? 
1. Something like func(ctx Context, topic string, payload []byte), since you are creating and passing a context to the lower level anyway.
2. Return an error instead of log.Fatal crash the process.

Reply:
@zzzming +1

Reply:
It's unclear what should happen here. Returning an error is not how the rest of the framework works. If there's an error, it dies, relying on metrics/backpressure to alert and remediate. If you look at the official documents for multi-topic dispatch (with python examples), they are not treated any differently as a unit. To me, this suggests they should be treated (and automatically tracked with metrics) the same way, otherwise the user should just create a regular consumer.